### PR TITLE
Recovery password

### PR DESCRIPTION
**Pull request by marcoacierno**
Request reset password mutation:

```
mutation {
  requestPasswordReset(input: {email: "eemail"}) {
    __typename
    
    ... on OperationResult {
      ok
    }
    
    
    ... on RequestPasswordResetMutationErrors {
      email
      nonFieldErrors
    }
  }
}
```

### CODE REVIEW

File: backend/users/models.py [hunk 1]
```diff
@@ -80,8 +80,6 @@ class User(AbstractBaseUser, PermissionsMixin):
     USERNAME_FIELD = "email"
     REQUIRED_FIELDS = []
 
-    objects = UserManager()
```
Comment (marcoacierno) [inline backend/users/models.py]: This was duplicated, there is an identical one a bit above it

File: backend/notifications/static/logo-pycon.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...7.85-10.95-31.31-10.95-13.35 0-23.73 3.68-31.14 11.03-7.41 7.36-11.12 17.6-11.12 30.71 0 13.23 3.71 23.5 11.12 30.8 7.41 7.3 17.85 10.95 31.31 10.95 13.35 0 23.72-3.68 31.14-11.03 7.41-7.35 11.12-17.59 11.12-30.71zM771.449 287.49l-70.66-85.37v85.37h-26.18V165.33h24.98l70.67 85.37v-85.37h26.17v122.16h-24.98zM833.06 202.47c-8.95-10.68-13.42-23.06-13.42-37.14 0-14.08 4.47-26.46 13.42-37.13h16.53c-4.1 6-7.1 11.98-9 17.96-1.9 5.97-2.86 12.37-2.86 19.18 0 6.81.95 13.2 2.86 19.18 1.91 5.97 4.91 11.96 9 17.97h-16.53v-.02zM858.08 191.74v-61.81h18.18v61.81h-18.18zM907.429 191.74v-46.92h-23.03v-14.89h64.23v14.89h-23.03v46.92h-18.17zM951.06 202.47c4.1-6 7.1-11.99 9-17.97 1.91-5.97 2.86-12.36 2.86-19.18 0-6.81-.95-13.2-2.86-19.18-1.91-5.97-4.91-11.96-9-17.96h16.53c8.95 10.67 13.42 23.05 13.42 37.13 0 14.08-4.47 26.46-13.42 37.14h-16.53v.02z" fill="#000"></path><path d="M1491.5 8.5h-433v433h433V8.5z" fill="#9473B0"></path><path d="M1483 17v416h-416V17h416zm17-17h-450v450h450V0z" fill="#000"></path><path d="M1210.97 285.06V192.6l-28.87 17.71-12.04-19.13 44.81-28.52h23.2v122.39h-27.1v.01zM1334.96 285.06V192.6l-28.87 17.71-12.04-19.13 44.81-28.52h23.2v122.39h-27.1v.01z" fill="#1D1D1B"></path></svg>
```
Comment (marcoacierno) [inline backend/notifications/static/logo-pycon.svg]: I have to remove this

File: backend/notifications/templates/notifications/simple_email.html [hunk 3]
```diff
@@ -0,0 +1,1297 @@
+<!doctype html>
```
Comment (marcoacierno) [inline backend/notifications/templates/notifications/simple_email.html]: Generated from mailchimp, I need to use the static files for the images/etc.

File: backend/api/types.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import strawberry
+
+
+@strawberry.type
+class OperationResult:
```
Comment (marcoacierno) [inline backend/api/types.py]: The idea was to have a simple "type" when the mutation doesn't return any information more than "operation completed with success/received"

File: backend/api/users/forms.py [hunk 5]
```diff
@@ -1,12 +1,59 @@
...
+        token = cleaned_data["token"]
+        userid = urlsafe_b64decode(cleaned_data["b64UserId"])
+
+        user = User.objects.filter(id=userid).first()
+
+        if not user:
+            raise ValidationError({"b64UserId": _("Invalid user")})
+
+        if not default_token_generator.check_token(user, token):
+            raise ValidationError({"token": _("Invalid token")})
```
Comment (patrick91) [inline backend/api/users/forms.py]: TIL :)

File: backend/api/users/forms.py [hunk 6]
```diff
@@ -1,12 +1,59 @@
...
-
-from api.forms import ContextAwareModelForm
+from notifications.emails import send_request_password_reset_mail
 from strawberry_forms.forms import FormWithContext
 from users.models import User
 
 
+class ResetPasswordForm(FormWithContext):
+    token = CharField()
+    b64UserId = CharField()
```
Comment (patrick91) [inline backend/api/users/forms.py]:
```suggestion
    encoded_user_id = CharField()
```

:)

Comment (marcoacierno) [reply backend/api/users/forms.py]: done

File: backend/pycon/settings/prod.py [hunk 7]
```diff
@@ -26,3 +26,10 @@
 AWS_S3_REGION_NAME = env("AWS_REGION_NAME", default="eu-central-1")
 AWS_ACCESS_KEY_ID = env("AWS_ACCESS_KEY_ID", default=None)
 AWS_SECRET_ACCESS_KEY = env("AWS_SECRET_ACCESS_KEY", default=None)
+
+EMAIL_BACKEND = "django.core.mail.backends.smtp.EmailBackend"
+EMAIL_HOST = env("EMAIL_HOST", default=None)
+EMAIL_PORT = env("EMAIL_PORT", default=587)
+EMAIL_HOST_USER = env("EMAIL_HOST_USER", default=None)
+EMAIL_HOST_PASSWORD = env("EMAIL_HOST_PASSWORD", default=None)
+EMAIL_USE_TLS = env("EMAIL_USE_TLS", default=False)
```
Comment (patrick91) [inline backend/pycon/settings/prod.py]: we are going to use SES, so we should probably change these settings, let's remove them and
only keep  

```
EMAIL_BACKEND = env("EMAIL_BACKEND", default="django.core.mail.backends.locmem.EmailBackend")
```

in base.py :)

Comment (marcoacierno) [reply backend/pycon/settings/prod.py]: Done!

**Pull request merged**